### ISSUE TITLE
Under certain circumstances, NVDA can't find the "Notes for translators" window in Poedit

### ISSUE THREAD

**Issue by nvaccessAuto** [nvaccess/nvda#4371]
**Reported by k_kolev1985 on 2014-08-08 10:41**
I was doing some translations with Poedit today and I wanted to read the translators note for the current string. I pressed Control+Shift+A, but NVDA reported that it can't find the "Notes for translators" window. After some tests, I've found out that if the notification bar of Poedit is visible on the screen, NVDA couldn't find the "Notes for translators" window. If the notification bar of Poedit is not visible, NVDA will find and read the translators notes. The notification bar in Poedit is located above the list of translation strings and below the buttons toolbar. The notification bar doesn't appear always - only when there is an alert from Poedit (e.g. when the translation's project name isn't set). Some of the .po files for some NVDA add-ons have this problem and the issue can be reproduced with them (specially some .po files for description of add-ons for the site).

Steps to reproduce it:
1. Open a .po file in Poedit, witch has some problems (e.g. with no name for the project in the translation).
2. Press Control+Shift+A.

Actual results:
If the notification bar of Poedit is shown, NVDA reports that it can't find the "Notes for translators" window, even though it is visible on the screen.

Expected results:
NVDA should be able to find and read the "Notes for translators" window, no matter if the notification bar is shown or not.

Test environment:
- Operating System: Windows 8.1 Pro N, 64-bit, in bulgarian.
- Poedit version: 1.6.7, in english.
- NVDA versions: next-10966,a238c81 installed; 2014.2 portable.

**Comment by leonardder** [comment]
@k-kolev1985: Do you still have this issue with Poedit 1.x?

**Comment by k-kolev1985** [comment]
@leonardder Yes, I can still reproduce the issue with POEdit v1.6.11. NVDA was running with all add-ons disabled at the time.

**Comment by bhavyashah** [comment]
The current stable release of PoEdit is version 2.0.2. Given that this
version supercedes any outdated versions of the same software, does
the reported issue apply to this more recent version of PoEdit (which
is free software)?

On 7/21/17, Kostadin Kolev <notifications@github.com> wrote:
> @leonardder Yes, I can still reproduce the issue with POEdit v1.6.11. NVDA
> was running with all add-ons disabled at the time.
>
> --
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/nvaccess/nvda/issues/4371#issuecomment-316921656


-- 
Best Regards
Bhavya Shah

Blogger at Hiking Across Horizons: https://bhavyashah125.wordpress.com/

Contacting Me
E-mail Address: bhavya.shah125@gmail.com
Follow me on Twitter @BhavyaShah125 or www.twitter.com/BhavyaShah125
Mobile Number: +91 7506221750

**Comment by k-kolev1985** [comment]
Yes, the issue is reproducible even with the latest stable version (v2.0.2) of POEdit. But as all NVDA translators know, that latest version of POEdit is even less accessible than v1.6.

**Comment by vslavik** [comment]
> But as all NVDA translators know, that latest version of POEdit is even less accessible than v1.6.

Poedit developer here. As a matter of fact, a *lot* of work went into improving accessibility in v2, particularly in the underlying wx library.  It may be that "all NVDA translators know" that Poedit is utter crap, but not a single one of them bothered to file a [bug report](https://github.com/vslavik/poedit/issues), so maybe the situation is not as dire as that.

Or maybe it is, but in the case *telling the developer* is the necessary precondition to making it right.   If you encounter an a11y bug in Poedit, please file a bug. If there's some tweak that would make your life as NVDA developer easier, please file a bug. I can't promise to fix everything, there are practical limits imposed by wx for example, but I'm happy to do as much as I reasonably can to make NVDA's job easier. 

(P.S. Obligatory corrective remark, name's "Poedit" (as in normal English), not "POEdit" or "PoEdit" or any other permutation of casing.)

**Comment by zstanecic** [comment]
Hi václav,

You are not right, all depends on cooperation between users and developer.

People are telling the truth and that’s obvious, that poedit 1.6.10 has 
a lot of accessibility support, and this is the one of the last 
versions, wich supports screenreader enhancements.

If the state from this version can be reverted, it will be great.


W dniu 22.07.2017 o 10:55, Václav Slavík pisze:

**Comment by leonardder** [comment]
I belief some people are making wrong assumptions here. The fact that most of the utility functions (e.g. to report the comments and translator window) don't work in Poedit 2, isn't the fault of the Poedit developers. It is just because the switch to the new GUI made Poedit 2 incompatible with the current appModule.

**Comment by k-kolev1985** [comment]
@vslavik 
Hi Václav.

Thanks for answering.

I just yesterday realized that I could easily file a bug report to "Poedit" about accessibility (or anything else related to it for that matter), because that is when I saw that "Poedit" has an account on GitHub. Now that I know, I'll certainly file my bug reports there.

**Comment by bhavyashah** [comment]
@k-kolev1985 Could you please provide a status update on this issue? Also, if this still stands, please provide the link to the corresponding PoEdit ticket that you have filed so that others can track any progress made on this problem as well.

**Comment by k-kolev1985** [comment]
@bhavyashah Yes, I can still reproduce the issue with Poedit v1.6.11. I haven't tried it with the latest stable version of Poedit, because for me v1.6 is more comfortable to use as a translator. I may update and see if something has changed in Poedit recently.

And no - for now I haven't filed an issue in the GitHub page of Poedit, mainly because it is not clear to me who is responsible for this regression - NVDA or Poedit.

**Comment by k-kolev1985** [comment]
@Adriani90 Not exactly. I've filed a bug for Poedit on GitHub, but it was about TAB navigation, which includes the fields for comments and notes. Now the only way to read notes for translators in Poedit 2 is to use NVDA's object navigation to get to the object which is the text for the notes.

I don't have enough technical knowledge to explain to the developer of Poedit the bug about not being able to read the notes for translators with the provided by NVDA command Control+Shift+A. I don't mind filing the bug, if someone provides me the necessary technical knowledge/information for the issue.
